.github/workflows/test.yml

```diff
@@ -100,12 +104,17 @@ jobs:
             qemu-user-binfmt
 
       - name: Unit testing (for Linux-Arm64)
-        if: ${{ startsWith(runner.os, 'Linux') }}
+        if: ${{ startsWith(runner.os, 'Linux') && (success() || failure()) }}
         run: |
           GOOS=linux GOARCH=arm64 go test -v ./... > build/cbuildtests-${{ matrix.target }}-arm64.txt
+
+      - name: Generate JUnit test report (for Linux-Arm64)
+        if: ${{ startsWith(runner.os, 'Linux') && (success() || failure()) }}
+        run: |
           go-junit-report -set-exit-code -in build/cbuildtests-${{ matrix.target }}-arm64.txt -iocopy -out build/cbuild-testreport-${{ matrix.target }}-arm64.xml
 
       - name: Archive unit test results
+        if: always()
```

Comment:
If we do not run the report generation, we probably don't need to run the upload as well.

Reply:
[Addressed review comments](https://github.com/Open-CMSIS-Pack/cbuild/pull/78/commits/d8ebc2fd7bf590b252b1e25ed3b79709d5ca63d9)